helpers/release/main.go

```diff
@@ -0,0 +1,298 @@
+		nextVer.IncrementPatch()
+	}
+
+	fmt.Println()
+	fmt.Printf("✅ New version will be: %s\n", nextVer.String())
+	fmt.Println()
+	fmt.Println("❓ Do you want to continue? (press any key to continue or Ctrl+C to abort)")
+	fmt.Scanln()
+
+	// - update VERSION
+	if err := writeVersion(nextVer); err != nil {
+		panic(err)
+	}
+	fmt.Println("✅ Wrote VERSION")
+	time.Sleep(100 * time.Millisecond)
+	// - update version.go
+	if err := writeVersionGo(nextVer); err != nil {
+		panic(err)
+	}
+	fmt.Println("✅ Wrote version.go")
+	time.Sleep(100 * time.Millisecond)
+	// - update CHANGELOG.md
+	if err := writeChangelog(gitVer, nextVer); err != nil {
+		panic(err)
+	}
+	fmt.Println("✅ Updated CHANGELOG.md")
+	time.Sleep(100 * time.Millisecond)
+
+	// - create PR
+	//   git checkout -b release/vX.Y.Z
+	if err := gitCoRel(nextVer); err != nil {
+		panic(err)
+	}
+	fmt.Printf("✅ Created branch release/v%s\n", nextVer.String())
+	time.Sleep(100 * time.Millisecond)
+
+	// commit changes
+	if err := gitCommit(nextVer); err != nil {
+		panic(err)
+	}
+	fmt.Printf("✅ Committed changes to release/v%s\n", nextVer.String())
+	time.Sleep(100 * time.Millisecond)
+
+	fmt.Println("🏁 Done")
+	fmt.Printf("⚠ Prepared release of gopass %s. Now push this branch to your fork and open a pull request against gopasspw/gopass master.\n", nextVer.String())
+	fmt.Println()
+	fmt.Printf("⚠ After that PR has been merged tag it as 'v%s' and push that tag to kick of the release proecess.\n", nextVer.String())
+	fmt.Println()
+}
+
+func gitCoMaster() error {
+	cmd := exec.Command("git", "checkout", "master")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func gitPom() error {
+	cmd := exec.Command("git", "pull", "origin", "master")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func gitCoRel(v semver.Version) error {
+	cmd := exec.Command("git", "checkout", "-b", "release/v"+v.String())
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func gitCommit(v semver.Version) error {
+	cmd := exec.Command("git", "add", "CHANGELOG.md", "VERSION", "version.go")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	cmd = exec.Command("git", "commit", "-s", "-m", "Tag v"+v.String())
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func writeChangelog(prev, next semver.Version) error {
+	cl, err := changelogEntries(prev)
+	if err != nil {
+		panic(err)
+	}
+
+	fh, err := os.Create("CHANGELOG.new")
+	if err != nil {
+		return err
+	}
+	defer fh.Close()
+
+	fmt.Fprintf(fh, "## %s / %s\n\n", next.String(), time.Now().UTC().Format("2006-01-02"))
+	for _, e := range cl {
+		fmt.Fprint(fh, "* ")
+		fmt.Fprintln(fh, e)
+	}
+	fmt.Fprintln(fh)
+
+	ofh, err := os.Open("CHANGELOG.md")
+	if err != nil {
+		return err
+	}
+	defer ofh.Close()
+
+	if _, err := io.Copy(fh, ofh); err != nil {
+		return err
+	}
+
+	return os.Rename("CHANGELOG.new", "CHANGELOG.md")
```

Comment:
And yet you rename it, which overwrite it? Why bother with the Copy then?

Reply:
Because I need to prepend the new entries to the file.